hackathon_site/dashboard/frontend/src/components/cart/CartCard/CartCard.test.js

```diff
@@ -0,0 +1,85 @@
+import React from "react";
+import CartCard from "./CartCard";
+import { render } from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
+import { fireEvent } from "@testing-library/react";
+
+describe("CartCard Item test", () => {
+    const image = "https://i.imgur.com/IO6e5a6.jpg";
+    const title = "Some very long title and stuff";
+    const currentStock = 9;
+    const handleRemove = () => {};
+    const checkedOutQuantity = 3;
+    const outOfStock = "Currently unavailable";
+    const error = null;
+
+    test("CartCard default display test", () => {
+        const { getByText, getByAltText } = render(
+            <CartCard
+                image={image}
+                title={title}
+                currentStock={currentStock}
+                handleRemove={handleRemove}
+                checkedOutQuantity={checkedOutQuantity}
+                error={error}
+            />
+        );
+
+        expect(getByText(checkedOutQuantity.toString())).toBeInTheDocument();
+        expect(getByAltText(title)).toBeInTheDocument();
+        expect(getByText(title)).toBeInTheDocument();
+    });
+
+    test("Button that removes the CartCard item", () => {
+        const handleRemove = jest.fn();
```

Comment:
This works, but just be wary that you're playing with variable scope here since you already have `const handleRemove` defined outside (this applies for all of the tests where you've overriding something).

For future reference, you can make a function that renders the component with some default props, and then also accepts new ones to override them. Something like

```javascript
const renderComponent = (customProps) => {
    return render(<CartCard image={defaultImage} title="Some default title" {...customProps} />);
};
```

`customProps` needs to be an image, so you would wind up in your test doing something like:

```javascript
const { getByRole } = renderComponent({handleRemove: handleRemoveSpy});
```